src/common/time/TimeUtils.h

```diff
@@ -44,6 +44,14 @@ class TimeUtils {
     if ((p[date.month] - p[date.month - 1]) < date.day) {
       return Status::Error("`%s' is not a valid date.", date.toString().c_str());
     }
+    if (date.year < 0 || date.year > 9999) {
```

Comment:
Check it in parser, greater than `9999` is also valid, our upper bound is `int32_t::max()`.

Reply:
Lower bound too.

Reply:
I talk with @HarrisChu . 
He thinks `date.year` should be year. Should not be `int32_t::max()`.

Reply:
It's obviously incorrect, we should provide time range as wide as possible for user. You could check Postgres in https://www.postgresql.org/docs/9.1/datatype-datetime.html or other databases.

Reply:
> 

what's your expected for below queries:

```
yield datetime(-1)
yield datetime(1000)
yield datetime(123456)

yield datetime("-1")
yield datetime("1000")
yield datetime("123456")
```

Reply:
All valid date.

Reply:
Negative date means B.C.

Reply:
![image](https://user-images.githubusercontent.com/1726587/185534032-5cfc5f12-a27f-4629-8544-591e844cb5b9.png)

one is for epoch, the other is just year

Reply:
datetime("-1") is unexpected.

Reply:
fixed. year is in [-32767,32767].